src/o365/spo/commands/site/site-rename.ts

```diff
@@ -0,0 +1,285 @@
+import commands from '../../commands';
+import request from '../../../../request';
+import SpoCommand from '../../../base/SpoCommand';
+import { CommandOption, CommandValidate, CommandCancel } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import { FormDigestInfo} from '../../spo';
+
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  newSiteUrl: string;
+  newSiteTitle?: string;
+  suppressMarketplaceAppCheck?: boolean;
+  suppressWorkflow2013Check?: boolean;
+  wait?:boolean;
+}
+
+class SpoSiteRenameCommand extends SpoCommand {
+  private context?: FormDigestInfo;
+  private timeout?: NodeJS.Timer;
+  private operationData: any;
```

Comment:
`operationData` returns structured data so for type safety we should define a type for it.